src/Compatibility/Core/src/Android/AndroidAppIndexProvider.cs

```diff
@@ -1,10 +1,14 @@
 using System;
+using System.Diagnostics.CodeAnalysis;
 using Android.Content;
 
 namespace Microsoft.Maui.Controls.Compatibility.Platform.Android
 {
 	public class AndroidAppIndexProvider : IAppIndexingProvider
 	{
+		[UnconditionalSuppressMessage ("Trimming", "IL2035", Justification = AppLinksAssemblyName + ".dll is not always present.")]
+		[UnconditionalSuppressMessage ("Trimming", "IL2072", Justification = AppLinksAssemblyName + ".dll is not always present.")]
+		[DynamicDependency (DynamicallyAccessedMemberTypes.PublicConstructors, AppLinksAssemblyName + "." + AppLinksClassName, AppLinksAssemblyName)]
```

Comment:
Probably related to https://github.com/dotnet/linker/issues/2392 as well.
But currently this is likely the best way to solve this.